### DISCUSSION TITLE
Meshpy Triangle Point Attributes

### DISCUSSION THREAD

**Discussion by clockwork85** [inducer/meshpy#97] [Q&A] [answered]
Hello everyone, 

I am having issues understanding how to use point_attributes with MeshPy, similar to what I'm doing with Rufat's triangle wrapper with vertex attributes.  Meshpy's ability to pass a refinement function is something that I also need, and I don't think Rufat's library has this ability.  I can't figure out how, or where, to set point_attributes.  Here's a somewhat pseudo-code version of what I've tried to do:

```
from meshpy.triangle import build, MeshInfo
points3d = # some 3d numpy array of length 4310
points2d = points3d[:, :2] # some 2d numpy array of length 4310
elevation_attribs = points3d[:, 2] # some 1d numpy array of attributes - in this case z values for the points to create surface later
info = MeshInfo()
info.set_points(points2d) # can only set markers here
assert len(info.point_attributes) == len(points2d) # has correct length

info.point_attributes = elevation_attribs
# raises AttributeError: can't set attribute

info.point_attributes.resize(len(elevation_attribs))
# RuntimeError: sizes of slave arrays cannot be changed
```

Does anyone have any experience using point attributes with meshpy?  I am a bit at a loss, and I really need both the refinement function ability with point attributes.  

Thank you!

**Comment by inducer** [comment, marked as answer]
`test_point_attributes` may help:

https://github.com/inducer/meshpy/blob/aac52c49def3847b85bfa6a297dbc8f109e6b851/test/test_meshpy.py#L76-L90
